minicluster/src/main/java/tachyon/master/AbstractLocalTachyonCluster.java

```diff
@@ -0,0 +1,276 @@
+    testConf.set(Constants.WORKER_WORKER_BLOCK_THREADS_MIN, Integer.toString(1));
+    testConf.set(Constants.WORKER_WORKER_BLOCK_THREADS_MAX, Integer.toString(2048));
+    testConf.set(Constants.WORKER_NETWORK_NETTY_WORKER_THREADS, Integer.toString(2));
+
+    // Perform immediate shutdown of data server. Graceful shutdown is unnecessary and slow
+    testConf.set(Constants.WORKER_NETWORK_NETTY_SHUTDOWN_QUIET_PERIOD, Integer.toString(0));
+    testConf.set(Constants.WORKER_NETWORK_NETTY_SHUTDOWN_TIMEOUT, Integer.toString(0));
+
+    // Setup tiered store
+    String ramdiskPath = PathUtils.concatPath(mTachyonHome, "ramdisk");
+    testConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_ALIAS_FORMAT, 0), "MEM");
+    testConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_PATH_FORMAT, 0),
+        ramdiskPath);
+    testConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_QUOTA_FORMAT, 0),
+        Long.toString(mWorkerCapacityBytes));
+
+    int numLevel = testConf.getInt(Constants.WORKER_TIERED_STORE_LEVELS);
+    for (int level = 1; level < numLevel; level ++) {
+      String tierLevelDirPath =
+          String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_PATH_FORMAT, level);
+      String[] dirPaths = testConf.get(tierLevelDirPath).split(",");
+      List<String> newPaths = new ArrayList<String>();
+      for (String dirPath : dirPaths) {
+        String newPath = mTachyonHome + dirPath;
+        newPaths.add(newPath);
+      }
+      testConf.set(String.format(Constants.WORKER_TIERED_STORE_LEVEL_DIRS_PATH_FORMAT, level),
+              Joiner.on(',').join(newPaths));
+    }
+    return testConf;
+  }
+
+  protected void runWorker() throws IOException {
+    mWorker = new BlockWorker();
+    if (LineageUtils.isLineageEnabled(WorkerContext.getConf())) {
+      // Setup the lineage worker
+      LOG.info("Started lineage worker at worker with ID {}", WorkerIdRegistry.getWorkerId());
+      mLineageWorker = new LineageWorker(mWorker.getBlockDataManager());
+    }
+
+    Runnable runWorker = new Runnable() {
+      @Override
+      public void run() {
+        try {
+          // Start the lineage worker
+          if (LineageUtils.isLineageEnabled(WorkerContext.getConf())) {
+            mLineageWorker.start();
+          }
+          mWorker.process();
+
+        } catch (Exception e) {
+          throw new RuntimeException(e + " \n Start Worker Error \n" + e.getMessage(), e);
+        }
+      }
+    };
+    mWorkerThread = new Thread(runWorker);
+    mWorkerThread.start();
+  }
+
+  /**
+   * Returns a {@link tachyon.client.file.TachyonFileSystem} client.
+   *
+   * @return a TachyonFS client
+   * @throws IOException when the operation fails
+   */
+  public abstract TachyonFileSystem getClient() throws IOException;
+
+  public abstract int getMasterPort();
+
+  public TachyonConf getMasterTachyonConf() {
+    return mMasterConf;
+  }
+
+  protected void resetContext() {}
```

Comment:
Could you document all public and overridable methods? Then in the future, extensions of this class can know what's the design purpose of these methods.